snakemake/dag.py

```
@@ -2911,26 +2911,38 @@ def norm_rule_relpath(f, rule):
-                    "remote execution without shared filesystem):\n" + e.stderr.decode()
+        # check if a manifest file is provided
+        # or get git-managed files
+        if self.workflow.remote_execution_settings.source_file_manifest:
+            fl = self.workflow.remote_execution_settings.source_file_manifest
+            try:
+                with open(fl, "r") as manifest:
+                    for f in manifest.read().splitlines():
+                        if f.strip():
+                            files.add(os.path.relpath(f.strip()))
+            except FileNotFoundError:
+                raise WorkflowError(f"Manifest file not found: {fl}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)